tools/clang/test/CodeGenDXIL/hlsl/intrinsics/buffer-agg-load-stores.hlsl

```diff
@@ -0,0 +1,231 @@
+struct OffVector {
+  float4 pad1;
+  double pad2;
+  vector<T, N> v;
+  OffVector operator+(OffVector vec) {
+    OffVector ret;
+    ret.pad1 = 0.0;
+    ret.pad2 = 0.0;
+    ret.v = v + vec.v;
+    return ret;
+  }
+};
+
+  ByteAddressBuffer RoByBuf : register(t1);
+RWByteAddressBuffer RwByBuf : register(u1);
+
+  StructuredBuffer< TYNAME(TYPE,NUM) > RoStBuf : register(t2);
+RWStructuredBuffer< TYNAME(TYPE,NUM) > RwStBuf : register(u2);
+
+ConsumeStructuredBuffer< TYNAME(TYPE,NUM) > CnStBuf : register(u4);
+AppendStructuredBuffer< TYNAME(TYPE,NUM) > ApStBuf  : register(u5);
+
+TYPE Add(TYPE f1[NUM], TYPE f2[NUM])[NUM] {
+  TYPE ret[NUM];
+  for (int i = 0; i < NUM; i++)
+    ret[i] = f1[i] + f2[i];
+  return ret;
+}
+
+template<typename T>
+T Add(T v1, T v2) { return v1 + v2; }
+
+TYPE Add(TYPE f1[NUM], TYPE f2[NUM], TYPE f3[NUM], TYPE f4[NUM])[NUM] {
+  TYPE ret[NUM];
+  for (int i = 0; i < NUM; i++)
+    ret[i] = f1[i] + f2[i] + f3[i] + f4[i];
+  return ret;
+}
+
+template<typename T>
+T Add(T v1, T v2, T v3, T v4) { return v1 + v2 + v3 + v4; }
+
+void main(uint SIx[2] : SIX, uint1 VIx[2] : VIX) {
+  // ByteAddressBuffer Tests
+
+  // CHECK-DAG: [[HDLROBY:%.*]] = call %dx.types.Handle @dx.op.createHandleFromBinding(i32 217, %dx.types.ResBind { i32 1, i32 1, i32 0, i8 0 }, i32 1, i1 false)
+  // CHECK-DAG: [[HDLRWBY:%.*]] = call %dx.types.Handle @dx.op.createHandleFromBinding(i32 217, %dx.types.ResBind { i32 1, i32 1, i32 0, i8 1 }, i32 1, i1 false)
+
+  // CHECK-DAG: [[HDLROST:%.*]] = call %dx.types.Handle @dx.op.createHandleFromBinding(i32 217, %dx.types.ResBind { i32 2, i32 2, i32 0, i8 0 }, i32 2, i1 false)
+  // CHECK-DAG: [[HDLRWST:%.*]] = call %dx.types.Handle @dx.op.createHandleFromBinding(i32 217, %dx.types.ResBind { i32 2, i32 2, i32 0, i8 1 }, i32 2, i1 false)
+
+  // CHECK-DAG: [[HDLCON:%.*]] = call %dx.types.Handle @dx.op.createHandleFromBinding(i32 217, %dx.types.ResBind { i32 4, i32 4, i32 0, i8 1 }, i32 4, i1 false)
+  // CHECK-DAG: [[HDLAPP:%.*]] = call %dx.types.Handle @dx.op.createHandleFromBinding(i32 217, %dx.types.ResBind { i32 5, i32 5, i32 0, i8 1 }, i32 5, i1 false)
+
+  // CHECK-DAG: [[IX0:%.*]] = call i32 @dx.op.loadInput.i32(i32 4, i32 {{[0-9]*}}, i32 [[BOFF:0]]
+  // CHECK-DAG: [[RIX0:%.*]] = call i32 @dx.op.loadInput.i32(i32 4, i32 {{[0-9]*}}, i32 [[BOFF]]
+  
+
+  // CHECK: [[ANHDLRWBY:%.*]] = call %dx.types.Handle @dx.op.annotateHandle(i32 216, %dx.types.Handle [[HDLRWBY]]
+  // OFF: [[RIX0:%.*]] = add i32 [[IX0]], [[BOFF:[0-9]+]]
+  // CHECK: call %dx.types.ResRet.[[TY]] @dx.op.rawBufferLoad.[[TY]](i32 139, %dx.types.Handle [[ANHDLRWBY]], i32 [[RIX0]]
+  // MAT: [[IX0p4:%.*]] = add i32 [[RIX0]], [[p4:[0-9]+]]
+  // MAT: call %dx.types.ResRet.[[TY]] @dx.op.rawBufferLoad.[[TY]](i32 139, %dx.types.Handle [[ANHDLRWBY]], i32 [[IX0p4]]
+  // MAT: [[IX0p8:%.*]] = add i32 [[RIX0]], [[p8:[0-9]+]]
+  // MAT: call %dx.types.ResRet.[[TY]] @dx.op.rawBufferLoad.[[TY]](i32 139, %dx.types.Handle [[ANHDLRWBY]], i32 [[IX0p8]]
```

Comment:
The larger matrix types require multiple loads  because raw buffer loads are limited to 4 elements at a time. (for now). This checks for them.